### PR TITLE
Permanant delete stashed modification

### PR DESCRIPTION
**Pull request by lilian-houdelet**

### CODE REVIEW

File: src/components/dialogs/restore-modification-dialog.js [hunk 1]
```diff
@@ -52,16 +61,26 @@ const RestoreModificationDialog = ({
...
     const intl = useIntl();
-    const [modificationsToRestore, setModificationsToRestore] = useState([]);
 
-    useEffect(() => {
-        setModificationsToRestore(modifToRestore);
-    }, [modifToRestore]);
+    const [modificationsToRestore, setModificationsToRestore] = useState([]);
+    const [selectedItems, setSelectedItems] = useState(new Set());
+    const [toggleSelectAll, setToggleSelectAll] = useState(false);
+    const [openAlert, setOpenAlert] = useState(false);
```
Comment (souissimai) [inline src/components/dialogs/restore-modification-dialog.js:68]: const [popupConfirmDeletion, setPopupConfirmDeletion] = useState(false);

File: src/components/dialogs/restore-modification-dialog.js [hunk 2]
```diff
@@ -74,7 +93,13 @@ const RestoreModificationDialog = ({
         handleClose();
     };
 
-    const [selectedItems, setSelectedItems] = useState(new Set());
+    const handleSelectAll = useCallback(() => {
+        setToggleSelectAll((prev) => !prev);
```
Comment (souissimai) [inline src/components/dialogs/restore-modification-dialog.js:97]: setToggleSelectAllStashedModifications : suggestion to rename it

File: src/components/dialogs/restore-modification-dialog.js [hunk 3]
```diff
@@ -106,6 +131,23 @@ const RestoreModificationDialog = ({
                                 ref={provided.innerRef}
                                 {...provided.droppableProps}
                             >
+                                <Box sx={styles.selectAll}>
+                                    <Checkbox
+                                        color={'primary'}
+                                        edge="start"
+                                        checked={
+                                            selectedItems.size ===
+                                            modificationsToRestore.length
```
Comment (souissimai) [inline src/components/dialogs/restore-modification-dialog.js:140]: I believe a renaming is in order, given that we can now select stashed modifications to either restore or permanently delete them. My suggestion is to replace it by "stashedModifications"

File: src/components/dialogs/restore-modification-dialog.js [hunk 4]
```diff
@@ -52,16 +61,26 @@ const RestoreModificationDialog = ({
...
-    useEffect(() => {
-        setModificationsToRestore(modifToRestore);
-    }, [modifToRestore]);
+    const [modificationsToRestore, setModificationsToRestore] = useState([]);
+    const [selectedItems, setSelectedItems] = useState(new Set());
+    const [toggleSelectAll, setToggleSelectAll] = useState(false);
+    const [openAlert, setOpenAlert] = useState(false);
 
     const handleClose = () => {
         onClose();
```
Comment (souissimai) [inline src/components/dialogs/restore-modification-dialog.js:71]:
```suggestion
    const handleDelete = () => {
        setOpenAlert(false);
        const selectedModificationsUuidToDelete = [
            ...selectedItems.values(),
        ].map((item) => item.uuid);
        deleteModifications(
            studyUuid,
            currentNode.id,
            selectedModificationsUuidToDelete
        );
        handleClose();
    };```

File: src/components/dialogs/restore-modification-dialog.js [hunk 5]
```diff
@@ -52,16 +61,26 @@ const RestoreModificationDialog = ({
...
     };
 
+    const handleDelete = () => {
```
Comment (souissimai) [inline src/components/dialogs/restore-modification-dialog.js:74]: const selectedModificationsUuidsToDelete = [
            ...selectedItems.values(),
        ].map((item) => item.uuid);

**Review by souissimai — APPROVED** [review]
test ok code ok

**Review by ayolab — CHANGES_REQUESTED** [review]
there is some warnings that need to be fixed

**Review by ayolab — APPROVED** [review]

**Pull request merged**